pkg/conference/matrix_message_processor.go

```diff
@@ -53,7 +53,21 @@ func (c *Conference) onNewParticipant(participantID ParticipantID, inviteEvent *
 		messageSink := common.NewMessageSink(participantID, c.peerMessages)
 
 		keepAliveDeadline := time.Duration(c.config.KeepAliveTimeout) * time.Second
-		peer, answer, err := peer.NewPeer(inviteEvent.Offer.SDP, messageSink, logger, keepAliveDeadline)
+		pingInterval := time.Duration(c.config.PingInterval) * time.Second
+		sendPing := func() {
+			participant.sendDataChannelMessage(event.Event{
+				Type:    event.FocusCallPing,
+				Content: event.Content{},
+			})
+		}
+		peer, answer, err := peer.NewPeer(
+			inviteEvent.Offer.SDP,
+			messageSink,
+			logger,
+			pingInterval,
+			keepAliveDeadline,
+			sendPing,
```

Comment:
It seems like we're passing 3 parameters that are only related to the keep-alive/heartbeat logic. I think it would probably look a bit easier to read and maintain if we maybe group them together in a structure or something.

I also see that now we construct one of the closures inside the `Peer` as it was before, but another one inside `Conference`. Maybe in this case it would make sense to move the keep-alive handling to the conference level.

Or, pass the `HeartbeatConfiguration` or something like this to the `Peer`, so that the `Peer` just starts a go-routine passing a single parameter to the otherwise opaque structure? This would also mean we would not need to store keep-alive related vars inside `Peer`.

We could start the go-routine once `Peer` is constructed (like we did before) so that if the data channel is not connected within the deadline, we would treat it as hangup (also like we did before), which probably makes sense.